tortoisebeacon/weakcoin/weak_coin.go

```diff
@@ -131,41 +152,49 @@ func (wc *weakCoin) PublishProposal(ctx context.Context, epoch types.EpochID, ro
 		return fmt.Errorf("broadcast weak coin message: %w", err)
 	}
 
+	wc.Log.With().Info("Published proposal",
+		log.Uint64("epoch_id", uint64(epoch)),
+		log.Uint64("round_id", uint64(round)),
+		log.String("proposal", types.BytesToHash(signedProposal).ShortString()))
+
 	return nil
 }
 
 func (wc *weakCoin) proposalExceedsThreshold(proposal []byte) bool {
-	proposalInt := new(big.Int).SetBytes(proposal[:])
+	proposalInt := new(big.Int).SetBytes(proposal)
 
 	return proposalInt.Cmp(wc.threshold) == 1
 }
 
 func (wc *weakCoin) generateProposal(epoch types.EpochID, round types.RoundID) ([]byte, error) {
-	msg := bytes.Buffer{}
+	proposal := bytes.Buffer{}
 
-	msg.WriteString(proposalPrefix)
+	proposal.WriteString(proposalPrefix)
 
 	epochBuf := make([]byte, 8)
 	binary.LittleEndian.PutUint64(epochBuf, uint64(epoch))
-	if _, err := msg.Write(epochBuf); err != nil {
-		return nil, err
+	if _, err := proposal.Write(epochBuf); err != nil {
+		return nil, fmt.Errorf("write epoch to buffer: %w", err)
```

Comment:
please panic here. it shouldn't happen with buffer

Reply:
Yes, panic looks better here